checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsChecker.java

```diff
@@ -0,0 +1,94 @@
+package org.checkerframework.checker.calledmethods;
+
+import java.util.LinkedHashSet;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.returnsreceiver.ReturnsReceiverChecker;
+import org.checkerframework.common.value.ValueChecker;
+import org.checkerframework.framework.qual.StubFiles;
+import org.checkerframework.framework.source.SupportedOptions;
+import org.checkerframework.framework.source.SuppressWarningsPrefix;
+
+/**
+ * The primary typechecker for the Called Methods checker, which allows programmers to specify
+ * unsafe combinations of options to builder or builder-like interfaces and prevent dangerous
+ * objects from being instantiated.
+ *
+ * <p>Note that this checker respects the suppress warning keys from its earlier names ("builder",
+ * "object.construction", and "objectconstruction" in addition to its own keys ("calledmethods" and
+ * "called.methods").
+ */
+@SuppressWarningsPrefix({
+    "builder",
+    "object.construction",
+    "objectconstruction",
+    "calledmethods",
+    "called.methods"
+})
+@SupportedOptions({
+    CalledMethodsChecker.USE_VALUE_CHECKER,
+    CalledMethodsChecker.COUNT_FRAMEWORK_BUILD_CALLS,
+    CalledMethodsChecker.DISABLED_FRAMEWORK_SUPPORTS,
+})
+@StubFiles({"DescribeImages.astub", "GenerateDataKey.astub"})
+public class CalledMethodsChecker extends BaseTypeChecker {
+
+    /**
+     * If this option is supplied, use the Value Checker to reduce false positives when analyzing
+     * calls to the AWS SDK.
+     */
+    public static final String USE_VALUE_CHECKER = "useValueChecker";
+
+    /**
+     * If this option is supplied, count the number of analyzed calls to build() in supported
+     * frameworks and print it when analysis is complete. Useful for collecting metrics.
+     */
+    public static final String COUNT_FRAMEWORK_BUILD_CALLS = "countFrameworkBuildCalls";
```

Comment:
Do we still want to keep this code for collecting metrics?

Reply:
I don't see any reason to remove it

Reply:
Ok